### PR TITLE
COL-23 - Set up core Collabosphere infrastructure

### PR DESCRIPTION
**Pull request by nicolaasmatthijs**
- Basic module structure
- Express app server
- Logging
- Configuration
- Travis build
- Test framework

### CODE REVIEW

**Comment by nicolaasmatthijs** [comment]
I'll be cleaning this up a bit more, but this is ready for an initial review pass

File: gulpfile.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+gulp.task('mocha', function() {
+  return gulp
+    .src(['node_modules/col-tests/lib/beforeTests.js', 'node_modules/col-*/tests/**/*.js'])
+    .pipe(mocha());
+});
+
+/**
+ * Run the full Collabosphere test suite
+ */
+gulp.task('test', function() {
```
Comment (simong) [inline gulpfile.js]: Can we make this the default gulp task?

File: node_modules/col-core/lib/server.js [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+  app.set('json spaces', 0);
+
+  // Don't output the x-powered-by header
+  app.set('x-powered-by', false);
+
+  /*!
+   * Collabosphere supports the following type of request encodings:
+   *
+   *  * urlencoded (regular POST requests)
+   *  * application/json
```
Comment (simong) [inline node_modules/col-core/lib/server.js]: Is there a need for a JSON or multipart parser? It seems like most of the API interactions with this tool should be doable through plain-old urlencoded requests right?

**Comment by simong** [comment]
LGTM, are you planning on removing that dummy service and putting something more tangible before you want to get this merged?

File: gulpfile.js (hunk 1 above)
Comment (nicolaasmatthijs) [inline gulpfile.js]: I was thinking about making the production build the default task

File: node_modules/col-core/lib/server.js (hunk 2 above)
Comment (nicolaasmatthijs) [inline node_modules/col-core/lib/server.js]: Multipart will be necessary, as students will be able to upload files to the asset library. We can probably take out the application/json encoding though.

**Comment by nicolaasmatthijs** [comment]
@simong : I was thinking I would just leave the dummy service to give people a better idea of how it will work. As soon as the first service is in, we can take it out.

**Comment by nicolaasmatthijs** [comment]
Followed up on review. Back to @simong for review.

File: node_modules/col-core/lib/server.js (hunk 2 above)
Comment (simong) [inline node_modules/col-core/lib/server.js]: I can't recall the designs immediately but would they upload into the tool directly or as part of an assignment?

Comment (nicolaasmatthijs) [inline node_modules/col-core/lib/server.js]: Both

**Pull request merged**